tests/tests_insights_tags.yml

```diff
@@ -0,0 +1,110 @@
+    - name: Skip when running with embedded Candlepin
+      meta: end_play
+      when:
+        - lookup('env', 'LSR_RHC_TEST_DATA') | length == 0
+      delegate_to: 127.0.0.1
+      become: false
+
+    - name: Import test data
+      include_vars:
+        file: "{{ lookup('env', 'LSR_RHC_TEST_DATA') }}"
+      delegate_to: 127.0.0.1
+      become: false
+
+    - name: Configure tags and register insights
+      include_role:
+        name: linux-system-roles.rhc
+      vars:
+        rhc_auth:
+          login:
+            username: "{{ lsr_rhc_test_data.reg_username }}"
+            password: "{{ lsr_rhc_test_data.reg_password }}"
+        rhc_insights:
+          state: present
+          tags:
+            group: ansible
+            activity: test
+        rhc_organization: "{{ lsr_rhc_test_data.reg_organization }}"
+        rhc_server:
+          hostname: "{{ lsr_rhc_test_data.candlepin_host }}"
+          port: "{{ lsr_rhc_test_data.candlepin_port }}"
+          prefix: "{{ lsr_rhc_test_data.candlepin_prefix }}"
+          insecure: "{{ lsr_rhc_test_data.candlepin_insecure }}"
+        rhc_baseurl: "{{ lsr_rhc_test_data.baseurl | d(omit) }}"
+
+    - name: Get state of insights tags.yml file
+      include_tasks: get_insights_tags.yml
+
+    - name: Rename the tags to test_insights_tags_modified
+      set_fact:
+        test_insights_tags_new: "{{ test_insights_tags }}"
+
+    - name: Check if tags.yml file exists and its size is > 0
+      assert:
+        that: test_insights_tags_new.stat.size > 0
+
+    - name: Change tags
+      include_role:
+        name: linux-system-roles.rhc
+      vars:
+        rhc_insights:
+          tags:
+            group:
+              - ansible
+              - rhc
+
+    - name: Get state of insights tags.yml file
+      include_tasks: get_insights_tags.yml
+
+    - name: Rename the tags to test_insights_tags_modified
+      set_fact:
+        test_insights_tags_modified: "{{ test_insights_tags.stat.size }}"
+
+    - name: Check if tags.yml file exists and its size has changed
+      assert:
+        that:
+          - test_insights_tags_new.stat.size != test_insights_tags_modified
+
+    - name: Do nothing
+      include_role:
+        name: linux-system-roles.rhc
+      vars:
+        rhc_insights:
+          tags:
+            group:
+              - ansible
+              - rhc
+
+    - name: Get state of insights tags.yml file
+      include_tasks: get_insights_tags.yml
+
+    - name: Check if tags.yml file exists and its size is the same
+      assert:
+        that:
+          - test_insights_tags.stat.size == test_insights_tags_modified
+
+    - name: Remove all tags
+      include_role:
+        name: linux-system-roles.rhc
+      vars:
+        rhc_insights:
+
+    - name: Get state of insights tags.yml file
+      include_tasks: get_insights_tags.yml
+
+    - name: Check that insights tags.yml file does not exists
+      assert:
+        that: test_insights_tags.stat.exists == False
```

Comment:
or simply "not var" ;)

Reply:
I have tried this, but this var is registered so it doesn't work. I'll keep this.